src/pages/settings/e-invoice/peppol/Onboarding.tsx

```diff
@@ -0,0 +1,427 @@
+        <div>
+          {t('peppol_whitelabel_warning')} <br />
+          <Link to="/settings/account_management">{t('purchase_license')}</Link>
+        </div>
+      ) : null}
+
+      {isHosted() ? (
+        <div>
+          {t('peppol_plan_warning')} <br />
+          <Link to="/settings/account_management">
+            {t('pro_plan_call_to_action')}
+          </Link>
+        </div>
+      ) : null}
+
+      <div className="flex justify-end">
+        <Button behavior="button" type="primary" onClick={() => onContinue()}>
+          {t('continue')} (in dev)
+        </Button>
+      </div>
+    </div>
+  );
+}
+
+function Token({ onContinue }: StepProps) {
+  const { t } = useTranslation();
+
+  const accentColor = useAccentColor();
+
+  const [isTokenGenerated, setIsTokenGenerated] = useState(false);
+
+  return (
+    <div>
+      <p>{t('peppol_tokens_info')}</p>
+      <p className="mt-2">
+        Token is used as another step to make sure invoices are sent securely.
+        Unlike white-label licenses, token can be rotated at any point without
+        need to wait on Invoice Ninja support.
+      </p>
+
+      <div className="flex items-center gap-1 my-3">
+        <p>You need to generate a token to continue.</p>
+        <button
+          type="button"
+          style={{ color: accentColor }}
+          onClick={() => setIsTokenGenerated(true)}
+        >
+          Generate token
+        </button>
+      </div>
+
+      <div className="flex justify-end">
+        {isTokenGenerated ? (
+          <Button behavior="button" type="primary" onClick={() => onContinue()}>
+            {t('continue')}
+          </Button>
+        ) : null}
+      </div>
+    </div>
+  );
+}
+
+function BuyCredits({ onContinue }: StepProps) {
+  const { t } = useTranslation();
+
+  return (
+    <div>
+      <p>{t('peppol_credits_info')}</p>
+
+      <Link to="https://invoiceninja.com" external>
+        {t('buy_credits')}
+      </Link>
+
+      <div className="flex justify-end">
+        <Button behavior="button" type="primary" onClick={() => onContinue()}>
+          {t('continue')}
+        </Button>
+      </div>
+    </div>
+  );
+}
+
+function Form({ onContinue }: StepProps) {
+  const { t } = useTranslation();
+  const company = useCurrentCompany();
+  const refresh = useRefreshCompanyUsers();
+
+  const [errors, setErrors] = useState<ValidationBag | null>(null);
+
+  const form = useFormik({
+    initialValues: {
+      party_name: company?.settings?.name || '',
+      line1: company?.settings?.address1 || '',
+      line2: company?.settings?.address2 || '',
+      city: company?.settings?.city || '',
+      county: company?.settings?.state || '',
+      zip: company?.settings?.postal_code || '',
+      country: company?.settings?.country_id || '',
+    },
+    onSubmit: (values, { setSubmitting }) => {
+      toast.processing();
+
+      setErrors(null);
+
+      request('POST', endpoint('/api/v1/einvoice/peppol/setup'), values)
```

Comment:
I'm not sure we'll be able to proxy self hosted to the same route due to the guards in place?